run-tests.py

```diff
@@ -41,25 +43,44 @@ def cli():
 @cli.command(help="Build a container")
 @image_option
 @salt_option
-def build(image, salt):
-    dockerfile = "test/{0}.Dockerfile".format(image)
-    tag = get_tag(image, salt)
-    if salt:
-        dockerfile_content = open(dockerfile, "rb").read()
-        dockerfile_content += (
-            b"\n"
-            b"ADD test/minion.conf /etc/salt/minion.d/minion.conf\n"
-            b"ADD test/salt /srv/salt\n"
-            b"ADD test/pillar /srv/pillar\n"
-            b"ADD {0} /srv/formula/{0}\n"
-            b"RUN salt-call --hard-crash --retcode-passthrough -l debug state.highstate\n"
-        ).format(_formula)
-        dockerfile = os.path.join("test", "{0}_salted.Dockerfile".format(image))
-        with open(dockerfile, "wb") as fd:
-            fd.write(dockerfile_content)
-    subprocess.check_call([
-        "docker", "build", "-t", tag, "-f", dockerfile, ".",
-    ])
+@click.option('--tag', default=None)
+@click.option('--pillar-root', type=click.Path(exists=True), default='test/pillar')
+@click.option('--file-root', type=click.Path(exists=True), default='test/salt')
+@click.option('--docker-add-cmd', type=click.UNPROCESSED, default=None, help=(
+    "Custom docker instruction to add when building the image (eg. EXPOSE 8080)"
+))
+def build(image, salt, tag, pillar_root, file_root, docker_add_cmd):
+    dockerfile = tempfile.NamedTemporaryFile(
```

Comment:
`NamedTemporaryFile` can be used as a context manager, which handles deletion upon exit